### PR TITLE
feat: add support for pdf a

### PR DESCRIPTION
**Pull request by prog-r-amer**
Hi,

my first rust code ever, if there's something to improve please tell me.
I needed the Compiler option for PDFA2 from typst.

### CODE REVIEW

File: packages/typst.node/src/lib.rs [hunk 1]
```diff
@@ -134,11 +134,23 @@ pub struct QueryDocArgs {
     pub field: Option<String>,
 }
 
+#[napi(string_enum)]
+#[derive(serde::Serialize, serde::Deserialize, Debug, PartialEq, Eq)]
+pub enum NodePdfStandard {
```
Comment (Myriad-Dreamin) [inline packages/typst.node/src/lib.rs:139]: This enum seems too complicated. We can simply pass string option instead. From typst's definition, we can deserialize a pdf standard by `let standard: PdfStandard = serde_json::from_value(standard_option)`.

https://github.com/typst/typst/blob/5c37a1cfea2a08b08f4af34ba867748fd90c687e/crates/typst-pdf/src/lib.rs#L140

**Review by Myriad-Dreamin — COMMENTED** [review]
Thank you for submitting the PR. Overall, I don't see much problem inside.

**Comment by Myriad-Dreamin** [comment]
> Isn't the enum still the better way cause it's typed? I think that's pretty helpful.

I'm not pretty like to force the typing here, as we might have `v1.8`, `v1.9` and more in future. They sounds not really a list that can be exhausted.


Instead we can provide an enum but I think it is better to be provided as string enum at js/ts side. I believe they would be convenient as well:

```ts
enum PdfStandard {
    V_1_7 = "v1.7",
    A_2b = "A2b",
}
```

**Comment by prog-r-amer** [comment]
Hi, I updated the pull request. I am not 100% sure if thats what you wanted but that's why there is a reviewer ;)

**Comment by Myriad-Dreamin** [comment]
> Hi, I updated the pull request. I am not 100% sure if thats what you wanted but that's why there is a reviewer ;)

Sure, I'll improve usability then and @ you in the later PR.

The final remaining thing is to fix the failed tests. You can run it on your repository's GitHub Action or manually run the command in local.

**Comment by prog-r-amer** [comment]
finally found the time and motivation to fix the pipeline stuff.

**Comment by Myriad-Dreamin** [comment]
I was planning to add my effort in a separated PR, but I find all added tests failed because of [79021d7](https://github.com/Myriad-Dreamin/typst.ts/pull/606/commits/79021d7eaa669f6bb5188af33f4491ef96a1de02).

**Pull request merged**